core/primitives/src/runtime/config.rs

```diff
@@ -26,6 +27,34 @@ pub struct RuntimeConfig {
     pub account_creation_config: AccountCreationConfig,
 }
 
+/// We start with a specific runtime config at the genesis, but we might want to
+/// do slight adjustments to it, depending on the protocol version.
+///
+/// This struct takes care of returning a config appropriate for particular
+/// protocol version.
+pub struct CurrentRuntimeConfig {
+    genesis_runtime_config: Arc<RuntimeConfig>,
+    with_lower_storage_cost: OnceCell<Arc<RuntimeConfig>>,
+}
+
+impl CurrentRuntimeConfig {
+    pub fn new(genesis_runtime_config: RuntimeConfig) -> Self {
+        Self {
+            genesis_runtime_config: Arc::new(genesis_runtime_config),
+            with_lower_storage_cost: OnceCell::new(),
```

Comment:
Alternatively just go ahead and construct the alternative configuration. It would increase memory usage if the protocol feature is not used but simplify things (and minimally speed things up) for the case of the feature being enabled.

Reply:
Yeah, that's an extremely reasonable suggestion, to say the least! There's really no need for any kind of laziness here.